### ISSUE TITLE
Setup.bat fails to complete. 

### ISSUE THREAD

**Issue by zeuszl1** [bmaltais/kohya_ss#858]
Running into an issue where Setup.Bat is force closing out. 

venv folder already exists, skipping creation...
Do you want to uninstall previous versions of torch and associated files before installing? Usefull if you are upgrading from torch 1.12.1 to torch 2.0.0 or if you are downgrading from torch 2.0.0 to torch 1.12.1.
[1] - Yes
[2] - No (recommanded for most)
Enter your choice (1 or 2): 1
**WARNING: Skipping xformers as it is not installed.
WARNING: Skipping torch as it is not installed.
WARNING: Skipping torchvision as it is not installed.
WARNING: Skipping triton as it is not installed.**
Please choose the version of torch you want to install:
[1] - v1 (torch 1.12.1) (Recommended)
[2] - v2 (torch 2.0.0) (Experimental)
Enter your choice (1 or 2): 1
**WARNING: Skipping bitsandbytes as it is not installed.**
The directory name is invalid.
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu116
**ERROR: Could not find a version that satisfies the requirement torch==1.12.1+cu116 (from versions: 2.0.0, 2.0.1)
ERROR: No matching distribution found for torch==1.12.1+cu116**

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip
Ignoring bitsandbytes: markers 'sys_platform == "darwin" or sys_platform == "linux"' don't match your environment
Ignoring gradio: markers 'sys_platform == "darwin"' don't match your environment
Ignoring tensorboard: markers 'sys_platform == "darwin"' don't match your environment
Ignoring huggingface-hub: markers 'sys_platform == "darwin"' don't match your environment
Processing d:\ai\kohya_ss
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting accelerate==0.18.0
  Using cached accelerate-0.18.0-py3-none-any.whl (215 kB)
Collecting albumentations==1.3.0
  Using cached albumentations-1.3.0-py3-none-any.whl (123 kB)
Collecting altair==4.2.2
  Using cached altair-4.2.2-py3-none-any.whl (813 kB)
Collecting bitsandbytes==0.35.0
  Using cached bitsandbytes-0.35.0-py3-none-any.whl (62.5 MB)
Collecting dadaptation==1.5
  Using cached dadaptation-1.5.tar.gz (8.3 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting diffusers[torch]==0.10.2
  Using cached diffusers-0.10.2-py3-none-any.whl (503 kB)
Collecting easygui==0.98.3
  Using cached easygui-0.98.3-py2.py3-none-any.whl (92 kB)
Collecting einops==0.6.0
  Using cached einops-0.6.0-py3-none-any.whl (41 kB)
Collecting ftfy==6.1.1
  Using cached ftfy-6.1.1-py3-none-any.whl (53 kB)
Collecting gradio==3.28.1
  Using cached gradio-3.28.1-py3-none-any.whl (17.3 MB)
Collecting lion-pytorch==0.0.6
  Using cached lion_pytorch-0.0.6-py3-none-any.whl (4.2 kB)
Collecting opencv-python==4.7.0.68
  Using cached opencv_python-4.7.0.68-cp37-abi3-win_amd64.whl (38.2 MB)
Collecting pytorch-lightning==1.9.0
  Using cached pytorch_lightning-1.9.0-py3-none-any.whl (825 kB)
Collecting safetensors==0.2.6
  Using cached safetensors-0.2.6-cp311-cp311-win_amd64.whl (268 kB)
Collecting tensorboard==2.10.1
  Using cached tensorboard-2.10.1-py3-none-any.whl (5.9 MB)
Collecting tk==0.1.0
  Using cached tk-0.1.0-py3-none-any.whl (3.9 kB)
Collecting toml==0.10.2
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting transformers==4.26.0
  Using cached transformers-4.26.0-py3-none-any.whl (6.3 MB)
Collecting voluptuous==0.13.1
  Using cached voluptuous-0.13.1-py3-none-any.whl (29 kB)
Collecting wandb==0.15.0
  Using cached wandb-0.15.0-py3-none-any.whl (2.0 MB)
Collecting fairscale==0.4.13
  Using cached fairscale-0.4.13.tar.gz (266 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting requests==2.28.2
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Collecting timm==0.6.12
  Using cached timm-0.6.12-py3-none-any.whl (549 kB)
Collecting huggingface-hub==0.13.3
  Using cached huggingface_hub-0.13.3-py3-none-any.whl (199 kB)
**ERROR: Could not find a version that satisfies the requirement tensorflow==2.10.1 (from versions: 2.12.0rc0, 2.12.0rc1, 2.12.0, 2.13.0rc0)
ERROR: No matching distribution found for tensorflow==2.10.1**

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip
**ERROR: xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl is not a supported wheel on this platform.**

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip
Copy D:\AI\kohya_ss\bitsandbytes_windows\cextension.py to D:\AI\kohya_ss\venv\Lib\site-packages\bitsandbytes
Copy D:\AI\kohya_ss\bitsandbytes_windows\libbitsandbytes_cpu.dll to D:\AI\kohya_ss\venv\Lib\site-packages\bitsandbytes
Copy D:\AI\kohya_ss\bitsandbytes_windows\libbitsandbytes_cuda116.dll to D:\AI\kohya_ss\venv\Lib\site-packages\bitsandbytes
**Copy D:\AI\kohya_ss\bitsandbytes_windows\main.py to D:\AI\kohya_ss\venv\Lib\site-packages\bitsandbytes\cuda_setup
File not found error: [Errno 2] No such file or directory: 'D:\\AI\\kohya_ss\\venv\\Lib\\site-packages\\bitsandbytes\\cuda_setup'**
'accelerate' is not recognized as an internal or external command,
operable program or batch file.
Press any key to continue . . .

**Comment by zeuszl1** [comment]
Still having issues after troubleshooting. If anyone has any idea what could be the issue.

**Comment by neuralnermal** [comment]
First time installing Kohya_ss having exact same errors except for this part:
> WARNING: Skipping xformers as it is not installed.
WARNING: Skipping torch as it is not installed.
WARNING: Skipping torchvision as it is not installed.
WARNING: Skipping triton as it is not installed.
>
Everything else is the same.

What baffles me is this part:
>[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip
>
because when I check my pip version, it already says I have 23.1.2
If I try to update anyway, it says I'm up to date

**Comment by neuralnermal** [comment]
Okay I have a scuffed solution that worked for me, maybe it will work for you too

My problem was I had multiple versions of Python installed over the years,
If you have just one version, guess this solution won't do you any good.

So every time I ran setup.bat it always briefly showed that it recommends Python 3.10.9
I ignored it originally because I knew I had that version already installed.
But since I couldn't think of any other solutions, I began uninstalling Python versions I didn't need
When I uninstalled 3.11.X versions and ran setup.bat again, 
I got a new error saying I don't have the C:\Python311 folder after answering two questions about Pytorch.
I don't exactly know why setup.bat was looking for Python311 folder and not Python310 folder,
but anyways I created a new folder, named it Python311, copied every file from Python 3.10.9 installation, and pasted them in.
After that, I ran setup.bat again, and it worked.

I still have no idea why it wanted python311 folder, when it said it recommends ver 3.10.9
But I guess deleting unwanted versions of python helped find the version it actually needed

**Comment by zeuszl1** [comment]
> Okay I have a scuffed solution that worked for me, maybe it will work for you too
> 
> 
> 
> My problem was I had multiple versions of Python installed over the years,
> 
> If you have just one version, guess this solution won't do you any good.
> 
> 
> 
> So every time I ran setup.bat it always briefly showed that it recommends Python 3.10.9
> 
> I ignored it originally because I knew I had that version already installed.
> 
> But since I couldn't think of any other solutions, I began uninstalling Python versions I didn't need
> 
> When I uninstalled 3.11.X versions and ran setup.bat again, 
> 
> I got a new error saying I don't have the C:\Python311 folder after answering two questions about Pytorch.
> 
> I don't exactly know why setup.bat was looking for Python311 folder and not Python310 folder,
> 
> but anyways I created a new folder, named it Python311, copied every file from Python 3.10.9 installation, and pasted them in.
> 
> After that, I ran setup.bat again, and it worked.
> 
> 
> 
> I still have no idea why it wanted python311 folder, when it said it recommends ver 3.10.9
> 
> But I guess deleting unwanted versions of python helped find the version it actually needed
> 
> 
> 
> 
> 
> 

I'm going to have to try this when I get home, before I never had this issue. After I had reset windows now I'm having it so we will see lol

**Comment by rhos** [comment]
According to this https://download.pytorch.org/whl/torch/ we have torch-1.12.0  versions up to 3.10 python.
So you can't really install torch-1.12.0 with python 3.11 (current version). 
**You can use virtual env with python 3.10 to get around this issue.**

There is torch-1.13.0 for python 3.11, but kohya explicitly uses 1.12

**Comment by zeuszl1** [comment]
Just checked right now, i also had Py 3.11 but dont even remember installing it lol

**Comment by zamp** [comment]
I ran into the same issue. For some reason bitsandbytes package won't get installed properly and shows up as a binary file.
I found a workaround though:
1. Open CMD as admin and run "pip install bitsandbytes --user"
2. Run it again (this time copy the "Requirement already satisfied" path)
3. Copy the bitsandbytes directory (from "Requirement already satisfied" path) and paste it in kohya_ss/venv/Lib/site-packages/
4. Run setup again (as normal user)
